pkg/migration-tool/utils/client.go

```diff
@@ -152,19 +249,29 @@ type PrompbResponse struct {
 	NumBytesUncompressed int
 }
 
+var prompbResponsePool = sync.Pool{New: func() interface{} { return new(PrompbResponse) }}
+
+func PutPrompbResponse(p *PrompbResponse) {
+	p.ID = 0
+	p.NumBytesUncompressed = 0
+	p.NumBytesCompressed = 0
+	p.Result.Timeseries = p.Result.Timeseries[:0]
```

Comment:
where is this reused I don't see a check for cap(Result.Timeseries) anywhere

Reply:
In planner.go `mergeSubSlabsToSlab()`. Why do we need a check for cap?

Reply:
Yes, we aren't reusing `p.Result.Timeseries`, since the p.Result is overwritten by the response that we receive from the PromQL query, so reusing is not possible here. Thank you.